### PR TITLE
Apply the new curated groups

### PR DESCRIPTION
**Pull request by lindycoder**
Those are category name/identifier, the pretty name is now defined on the server side.

### CODE REVIEW

File: scanners/boostsecurityio/codeql/rules.yaml [hunk 1]
```diff
@@ -545,15 +545,15 @@ rules:
...
+    group: top10-software-data-integrity-failures
     name: CWE-494
     pretty_name: CWE-494 - Download of Code Without Integrity Check
     description: The code downloads source code or an executable from a remote location and executes the code without sufficiently verifying the origin and integrity of the code.
     ref: https://codeql.github.com/codeql-query-help/
   CWE-UNKNOWN:
     categories:
     - ALL
-    group: CWE Top 25
+    group: ##### MISSING #####
```
Comment (fproulx-boostsecurity) [inline scanners/boostsecurityio/codeql/rules.yaml:556]:
```suggestion
    group: top10-insecure-design
```

File: scanners/boostsecurityio/codeql/rules.yaml [hunk 2]
```diff
@@ -177,7 +177,7 @@ rules:
     - cwe-352
     - boost-baseline
     - boost-hardened
-    group: CWE Top 25
+    group: top10-crypto-failures
```
Comment (fproulx-boostsecurity) [inline scanners/boostsecurityio/codeql/rules.yaml:180]:
```suggestion
    group: top10-broken-access-control
```

File: scanners/boostsecurityio/gosec/rules.yaml [hunk 3]
```diff
@@ -64,7 +64,7 @@ rules:
   G107:
     categories:
       - ALL
-    group: CWE Top 25
+    group: ##### MISSING #####
```
Comment (fproulx-boostsecurity) [inline scanners/boostsecurityio/gosec/rules.yaml:67]:
```suggestion
    group: top10-server-side-request-forgery
```

File: scanners/boostsecurityio/gosec/rules.yaml [hunk 4]
```diff
@@ -93,7 +93,7 @@ rules:
   G601:
     categories:
       - ALL
-    group: CWE Top 25
+    group: ##### MISSING #####
```
Comment (fproulx-boostsecurity) [inline scanners/boostsecurityio/gosec/rules.yaml:96]:
```suggestion
    group: top10-insecure-design
```

File: scanners/boostsecurityio/gosec/rules.yaml [hunk 5]
```diff
@@ -102,7 +102,7 @@ rules:
   G109:
     categories:
       - ALL
-    group: CWE Top 25
+    group: ##### MISSING #####
```
Comment (fproulx-boostsecurity) [inline scanners/boostsecurityio/gosec/rules.yaml:105]:
```suggestion
    group: top10-insecure-design
```

File: scanners/boostsecurityio/gosec/rules.yaml [hunk 6]
```diff
@@ -111,7 +111,7 @@ rules:
   G113:
     categories:
       - ALL
-    group: CWE Top 25
+    group: ##### MISSING #####
```
Comment (fproulx-boostsecurity) [inline scanners/boostsecurityio/gosec/rules.yaml:114]:
```suggestion
    group: top10-insecure-design
```

File: scanners/boostsecurityio/gosec/rules.yaml [hunk 7]
```diff
@@ -140,7 +140,7 @@ rules:
   G103:
     categories:
       - ALL
-    group: CWE Top 25
+    group: ##### MISSING #####
```
Comment (fproulx-boostsecurity) [inline scanners/boostsecurityio/gosec/rules.yaml:143]:
```suggestion
    group: top10-insecure-design
```

File: scanners/boostsecurityio/gosec/rules.yaml [hunk 8]
```diff
@@ -167,7 +167,7 @@ rules:
   G306:
     categories:
       - ALL
-    group: CWE Top 25
+    group: ##### MISSING #####
```
Comment (fproulx-boostsecurity) [inline scanners/boostsecurityio/gosec/rules.yaml:170]:
```suggestion
    group: top10-broken-access-control
```

File: scanners/boostsecurityio/gosec/rules.yaml [hunk 9]
```diff
@@ -196,7 +196,7 @@ rules:
   G106:
     categories:
       - ALL
-    group: CWE Top 25
+    group: ##### MISSING #####
```
Comment (fproulx-boostsecurity) [inline scanners/boostsecurityio/gosec/rules.yaml:199]:
```suggestion
    group: top10-id-authn-failures
```

File: scanners/boostsecurityio/gosec/rules.yaml [hunk 10]
```diff
@@ -275,7 +275,7 @@ rules:
   G303:
     categories:
       - ALL
-    group: CWE Top 25
+    group: ##### MISSING #####
```
Comment (fproulx-boostsecurity) [inline scanners/boostsecurityio/gosec/rules.yaml:278]:
```suggestion
    group: top10-insecure-design
```

File: scanners/boostsecurityio/gosec/rules.yaml [hunk 11]
```diff
@@ -294,7 +294,7 @@ rules:
   G110:
     categories:
       - ALL
-    group: CWE Top 25
+    group: ##### MISSING #####
```
Comment (fproulx-boostsecurity) [inline scanners/boostsecurityio/gosec/rules.yaml:297]:
```suggestion
    group: top10-insecure-design
```

File: scanners/boostsecurityio/gosec/rules.yaml [hunk 12]
```diff
@@ -303,7 +303,7 @@ rules:
   G114:
     categories:
       - ALL
-    group: CWE Top 25
+    group: ##### MISSING #####
```
Comment (fproulx-boostsecurity) [inline scanners/boostsecurityio/gosec/rules.yaml:306]:
```suggestion
    group: top10-security-misconfiguration
```

File: scanners/boostsecurityio/gosec/rules.yaml [hunk 13]
```diff
@@ -312,7 +312,7 @@ rules:
   G104:
     categories:
       - ALL
-    group: CWE Top 25
+    group: ##### MISSING #####
```
Comment (fproulx-boostsecurity) [inline scanners/boostsecurityio/gosec/rules.yaml:315]:
```suggestion
    group: top10-insecure-design
```

File: scanners/boostsecurityio/gosec/rules.yaml [hunk 14]
```diff
@@ -321,7 +321,7 @@ rules:
   G307:
     categories:
       - ALL
-    group: CWE Top 25
+    group: ##### MISSING #####
```
Comment (fproulx-boostsecurity) [inline scanners/boostsecurityio/gosec/rules.yaml:324]:
```suggestion
    group: top10-insecure-design
```

File: scanners/boostsecurityio/gosec/rules.yaml [hunk 15]
```diff
@@ -131,7 +131,7 @@ rules:
     categories:
       - ALL
       - cwe-200
```
Comment (fproulx-boostsecurity) [inline scanners/boostsecurityio/gosec/rules.yaml:133]:
```suggestion
      - cwe-200
      - cwe-489
```

File: scanners/boostsecurityio/native-scanner/rules.yaml [hunk 16]
```diff
@@ -1412,7 +1412,7 @@ rules:
       - boost-hardened # Not sure where to put this one...
     description: Checks for mishandling of highly compressed data with a very high
       compression ratio producing large output.
-    group: CWE Top 25
+    group: ##### MISSING #####
```
Comment (fproulx-boostsecurity) [inline scanners/boostsecurityio/native-scanner/rules.yaml:1415]:
```suggestion
    group: top10-insecure-design
```

File: scanners/boostsecurityio/native-scanner/rules.yaml [hunk 17]
```diff
@@ -1954,7 +1954,7 @@ rules:
       - boost-baseline
       - boost-hardened
     description: Ensure server side templates are validated
-    group: CWE Top 25
+    group: ##### MISSING #####
```
Comment (fproulx-boostsecurity) [inline scanners/boostsecurityio/native-scanner/rules.yaml:1957]:
```suggestion
    group: top10-injection
```

File: scanners/boostsecurityio/native-scanner/rules.yaml [hunk 18]
```diff
@@ -1967,7 +1967,7 @@ rules:
       - boost-hardened
     description: Checks for code issuing HTTP requests from the server-side that could
       lead to an Server-Side Request Forgery (SSRF) attack.
-    group: CWE Top 25
+    group: ##### MISSING #####
```
Comment (fproulx-boostsecurity) [inline scanners/boostsecurityio/native-scanner/rules.yaml:1970]:
```suggestion
    group: top10-server-side-request-forgery
```

File: scanners/boostsecurityio/native-scanner/rules.yaml [hunk 19]
```diff
@@ -1996,7 +1996,7 @@ rules:
       - cwe-295
       - cwe-297
     description: Checks for network functions with TLS verification disabled.
-    group: CWE Top 25
+    group: ##### MISSING #####
```
Comment (fproulx-boostsecurity) [inline scanners/boostsecurityio/native-scanner/rules.yaml:1999]:
```suggestion
    group: top10-id-authn-failures
```

File: scanners/boostsecurityio/semgrep/rules.yaml [hunk 20]
```diff
@@ -518,16 +518,16 @@ rules:
...
+    group: top10-software-data-integrity-failures
     name: CWE-494
     pretty_name: CWE-494 - Download of Code Without Integrity Check
     description: The code downloads source code or an executable from a remote location and executes the code without sufficiently verifying the origin and integrity of the code.
     ref: https://github.com/returntocorp/semgrep-rules/
   CWE-UNKNOWN:
     categories:
     - ALL
-    group: CWE Top 25
+    group: ##### MISSING #####
```
Comment (lindycoder) [inline scanners/boostsecurityio/semgrep/rules.yaml:529]: @fproulx-boostsecurity only this one remaining

Comment (fproulx-boostsecurity) [inline scanners/boostsecurityio/semgrep/rules.yaml:529]:
```suggestion
    group: top10-insecure-design
```

File: scanners/boostsecurityio/semgrep/rules.yaml [hunk 21]
```diff
@@ -150,7 +150,7 @@ rules:
     - cwe-352
     - boost-baseline
     - boost-hardened
-    group: CWE Top 25
+    group: top10-crypto-failures
```
Comment (fproulx-boostsecurity) [inline scanners/boostsecurityio/semgrep/rules.yaml:153]:
```suggestion
    group: top10-broken-access-control
```

**Review by fproulx-boostsecurity — APPROVED** [review]

**Pull request merged**